File: solutions/go/lasagna/lasagna.go
package lasagna

// Define the expected oven time in minutes
const OvenTime = 40

// RemainingOvenTime returns the remaining minutes based on the `actual` minutes already in the oven.
func RemainingOvenTime(actualMinutesInOven int) int {
	return 40 - actualMinutesInOven
}

const timePerLayer = 2

// PreparationTime calculates the time needed to prepare the lasagna based on the amount of layers.
func PreparationTime(numberOfLayers int) int {
	return timePerLayer * numberOfLayers
}

// ElapsedTime calculates the time elapsed cooking the lasagna. This time includes the preparation time and the time the lasagna is baking in the oven.
func ElapsedTime(numberOfLayers, actualMinutesInOven int) int {
	return PreparationTime(numberOfLayers) + actualMinutesInOven
}

File: solutions/lua/lasagna/lasagna.lua
local lasagna = {}

-- Define the expected oven time in minutes
lasagna["oven_time"] = 40

-- Calculate the remaining oven time in minutes
function lasagna.remaining_oven_time(actual_minutes_in_oven)
  return lasagna["oven_time"] - actual_minutes_in_oven
end

-- Calculate the preparation time in minutes
function lasagna.preparation_time(number_of_layers)
  local time_per_layer = 2
  return time_per_layer * number_of_layers
end

-- Calculate the elapsed working time in minutes
function lasagna.elapsed_time(number_of_layers, actual_minutes_in_oven)
  return lasagna.preparation_time(number_of_layers) + actual_minutes_in_oven
end

return lasagna
